errors.go

```diff
@@ -109,6 +124,40 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch() BatchError {
+	return &batcherrtype{}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
+	return len(b.errors) == 0
+}
```

Comment:
See this: https://github.com/aws/aws-sdk-go/blob/master/aws/awserr/error.go and this https://github.com/aws/aws-sdk-go/blob/b2dc98bb584e48b0f5f39c93110633173c5da43c/aws/awserr/types.go - maybe they can help you.